### DISCUSSION TITLE
Please help me to achieve wave effect in text.

### DISCUSSION THREAD

**Discussion by vinayITT** [fabricjs/fabric.js#9588] [Q&A]
![image](https://github.com/fabricjs/fabric.js/assets/113539654/35aae6e4-e516-4654-8e4e-39bab69ab348)
https://i.gyazo.com/3075d0e3cb8b8aeac41515b8774d9322.gif

I want to achieve **wave** effect on text as it is shown in shared media.
I don't have any idea how we can achieve such kind of effect on text.
I searched a lot but didn't found any way to achieve it.
Please show me some way to do it. 🙏

https://jsfiddle.net/iamuser003/3hf09apy/2/

**Comment by ShaMan123** [comment]
using fabric text you can override the `_renderChar` method and change the y value of the rendering context by applying `ctx.translate(0, animationValue)`

You can do other things like splitting the chars and creating a text object from each and using `animate`
Or you could look into webgl abilities like in pixi.js

**Reply by ShaMan123** [reply]
https://pixijs.com/examples/mesh-and-shaders/textured-mesh-advanced

**Comment by asturur** [comment]
There is a style property called deltaY that could be used for this without overriding

**Reply by vinayITT** [reply]
Yes, changing deltaY value and splitting the chars, and creating a text object will work.
But I am curious in case of some **custom animation and effects**, we may need WebGL integration.

**Can we integrate WebGL with fabric JS text elements without using any other libraries like pixi.js?**

It would be better if you can provide some code sample for that or any other references.
